File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/mission/port/UserMissionRepository.java
package com.study.webflux.rag.domain.mission.port;

import com.study.webflux.rag.domain.dialogue.model.UserId;
import com.study.webflux.rag.domain.mission.model.MissionId;
import com.study.webflux.rag.domain.mission.model.UserMission;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface UserMissionRepository {

	Flux<UserMission> findByUserId(UserId userId);

	Mono<UserMission> findByUserIdAndMissionId(UserId userId, MissionId missionId);

	Mono<UserMission> save(UserMission userMission);
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/mission/port/UserMissionRepository.kt
package com.miyou.app.domain.mission.port

import com.miyou.app.domain.dialogue.model.UserId
import com.miyou.app.domain.mission.model.MissionId
import com.miyou.app.domain.mission.model.UserMission
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface UserMissionRepository {
    fun findByUserId(userId: UserId): Flux<UserMission>

    fun findByUserIdAndMissionId(
        userId: UserId,
        missionId: MissionId,
    ): Mono<UserMission>

    fun save(userMission: UserMission): Mono<UserMission>
}
